src/lib/listeners/ProgressEventListener.ts

```diff
@@ -74,8 +74,8 @@ abstract class ProgressSpinner {
 
 export class RuleSelectionProgressSpinner extends ProgressSpinner implements ProgressEventListener {
 	private isListening: boolean;
-	private engineNames: string[];
-	private completionPercent: number;
+	private engineNames: Set<string>;
+	private readonly completionPercentages: number[];
```

Comment:
This is an array now, so it can track multiple percentages at once. I thought about having this be a `Map<CodeAnalyzer, number>`, which would properly support the case where the input array includes the same Core instance multiple times. But that's a pathological case anyway, so I decided that this implementation is good enough.